src/components/back-link-helper/back-link-helper.js

```diff
@@ -5,27 +5,14 @@ function BackLinkHelper($module, window, document) {
 }
 
 BackLinkHelper.prototype.init = function init() {
-  const isReferrerExternal = () => {
-    // If there is no referrer, consider it on the same domain
-    if (!this.document.referrer) {
-      return false;
-    }
-
-    try {
-      const currentDomain = new URL(this.window.location.href).hostname;
-      const referrerDomain = new URL(this.document.referrer).hostname;
-
-      // Check if the referrer is not the same as the current domain
-      return currentDomain !== referrerDomain;
-    } catch (err) {
-      return false;
-    }
-  };
-
   // do nothing if History API is absent
   if (this.window.history) {
-    // hide the backlink if the referrer is on a different domain
-    if (isReferrerExternal()) {
+    // TODO: Appsec have not yet given a clear answer on if we need to do the same domain check
```

Comment:
Do we want this TODO to reference today's discussion? There's the AppSec input, but also the UR consideration.